.github/workflows/release-and-update-changelog.yml

```diff
@@ -0,0 +1,52 @@
+name: Release and Update Changelog
+
+on:
+  push:
+    branches:
+      - feature/not-dev
+
+jobs:
+  release-and-update:
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Validate Commit Message
+        run: |
+          python scripts/validate_commit_message.py
+        env:
+          COMMIT_MESSAGE: ${{ github.event.head_commit.message }}
+
+      - name: Update the changelog
+        run: |
+          git config user.name "GitHub Action"
+          git config user.email "action@github.com"
+          python scripts/update_changelog.py
+
+      - name: Commit and push changelog changes
+        run: |
+          if ! git diff --quiet CHANGELOG.md; then
+            git config user.name "GitHub Action"
```

Comment:
Why not reuse the same action step - rather than having to repeat the git config?